Stride-Labs/stride
# [LSM] LSMLiquidStakeEnabled flag on host zone

## Context and purpose of the change
Since not all host zones will have LSM enabled, we need a flag on the host zone to indicate whether they have it enabled and whether LSM liquid stakes are supported.

## Brief Changelog
* Added bool field `LsmLiquidStakeEnabled` on host zone
* Added check in LSM liquid stake that it's enabled
* Extended register host zone command to include parameter

## Testing Instructions
* Start dockernet
```bash
make start-docker
```
* Confirm LSM is currently enabled by viewing the field `lsm_liquid_stake_enabled` on the host zone
```bash
strided q stakeibc list-host-zone
```
* Build a proposal file to disable LSM (call it `proposal.json`)
```json
{
    "description": "Proposal to disable LSM",
    "hostZone": "LSM",
    "enabled": false,
    "deposit": "10000001ustrd"
}
```

* Submit the proposal and vote. Make sure to vote before before the voting period ends (30 seconds)
```
strided tx gov submit-legacy-proposal toggle-lsm proposal.json --from val1 --gas auto -y 
strided tx gov vote 1 yes --from val1 -y
strided tx gov vote 1 yes --from val2 -y
strided tx gov vote 1 yes --from val3 -y
```

* Wait for the proposal to pass and then view the host zone again, now `lsm_liquid_stake_enabled` should be set to false
```bash
strided q stakeibc list-host-zone
```

* Update the proposal.json to the following (to re-enable LSM)
```json
{
    "description": "Proposal to enable LSM",
    "hostZone": "LSM",
    "enabled": true,
    "deposit": "10000001ustrd"
}
```
* Submit the proposal and vote
```
strided tx gov submit-legacy-proposal toggle-lsm proposal.json --from val1 --gas auto -y 
strided tx gov vote 2 yes --from val1 -y
strided tx gov vote 2 yes --from val2 -y
strided tx gov vote 2 yes --from val3 -y
```

* Wait for the proposal to pass and then view the host zone again, now `lsm_liquid_stake_enabled` should be set to true once again
```bash
strided q stakeibc list-host-zone
```
